client/extensions/woocommerce/state/products/selectors.js

```diff
@@ -0,0 +1,13 @@
+
+/**
+ * Gets product fetched from server.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} productId Numeric product Id.
+ * @return {Object} Product object from API.
+ */
+export function getProduct( state, productId ) {
```

Comment:
I think productId won't be enough since we will have multiple stores in the state, no?  In calypso's handling of posts, they ended up needing a concept of "globalPostID" and use memoizing to get to a siteID+postID https://github.com/Automattic/wp-calypso/blob/8fc98ea50609b8d5f8d0aa64b8146f5a0d8dd720/client/state/reader/posts/selectors.js#L24